derive_where(Debug) compile error if there's a struct parameter named `f`

```rust
use std::marker::PhantomData;
use derive_where::derive_where;

#[derive_where(Debug)]
pub struct Problematic<T> {
    f: String,
    _marker: PhantomData<T>,
}
```

results in
```txt
error[E0308]: mismatched types
    --> src/lib.rs:5:1
     |
5    | #[derive_where(Debug)]
     | ^^^^^^^^^^^^^^^^^^^^^^
     | |
     | types differ in mutability
     | arguments to this function are incorrect
     |
     = note: expected mutable reference `&mut Formatter<'_>`
                        found reference `&String`

```

The reason is that the expanded macro is trying to use a symbol `__f` in two conflicting ways: once for the struct member `f`, and once for the second argument to `Debug::fmt`. `cargo expand` shows that the example above expands to:

```rust
impl<T> ::core::fmt::Debug for Problematic<T> {
    fn fmt(&self, __f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
        match self {
            Problematic { f: ref __f, _marker: ref ___marker } => {
                let mut __builder = ::core::fmt::Formatter::debug_struct(
                    __f,
                    "Problematic",
                );
                ::core::fmt::DebugStruct::field(&mut __builder, "f", __f);
                ::core::fmt::DebugStruct::field(&mut __builder, "_marker", ___marker);
                ::core::fmt::DebugStruct::finish(&mut __builder)
            }
        }
    }
}
```

The first argument to `debug_struct` is our `String`, `f`, where it was meant to be the `&mut Formatter`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+- Avoid collisions between field names and trait method parameters.
+
 ### Changed
 - `PartialOrd` implementations now use `Ord` if applicable.
 

--- a/src/data/field.rs
+++ b/src/data/field.rs
@@ -110,8 +110,8 @@ impl<'a> Field<'a> {
 		type_: &'a Type,
 	) -> Result<Self> {
 		let attr = FieldAttr::from_attrs(derive_wheres, skip_inner, attrs)?;
-		let self_ident = format_ident!("__{}", member);
-		let other_ident = format_ident!("__other_{}", member);
+		let self_ident = format_ident!("__field_{}", member);
+		let other_ident = format_ident!("__other_field_{}", member);
 
 		Ok(Self {
 			attr,

--- a/src/test/basic.rs
+++ b/src/test/basic.rs
@@ -24,9 +24,9 @@ fn struct_() -> Result<()> {
 			impl<T> ::core::fmt::Debug for Test<T> {
 				fn fmt(&self, __f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
 					match self {
-						Test { field: ref __field } => {
+						Test { field: ref __field_field } => {
 							let mut __builder = ::core::fmt::Formatter::debug_struct(__f, "Test");
-							::core::fmt::DebugStruct::field(&mut __builder, "field", __field);
+							::core::fmt::DebugStruct::field(&mut __builder, "field", __field_field);
 							::core::fmt::DebugStruct::finish(&mut __builder)
 						}
 					}
@@ -52,7 +52,7 @@ fn struct_() -> Result<()> {
 			impl<T> ::core::hash::Hash for Test<T> {
 				fn hash<__H: ::core::hash::Hasher>(&self, __state: &mut __H) {
 					match self {
-						Test { field: ref __field } => { ::core::hash::Hash::hash(__field, __state); }
+						Test { field: ref __field_field } => { ::core::hash::Hash::hash(__field_field, __state); }
 					}
 				}
 			}
@@ -61,8 +61,8 @@ fn struct_() -> Result<()> {
 				#[inline]
 				fn cmp(&self, __other: &Self) -> ::core::cmp::Ordering {
 					match (self, __other) {
-						(Test { field: ref __field }, Test { field: ref __other_field }) =>
-							match ::core::cmp::Ord::cmp(__field, __other_field) {
+						(Test { field: ref __field_field }, Test { field: ref __other_field_field }) =>
+							match ::core::cmp::Ord::cmp(__field_field, __other_field_field) {
 								::core::cmp::Ordering::Equal => ::core::cmp::Ordering::Equal,
 								__cmp => __cmp,
 							},
@@ -74,8 +74,8 @@ fn struct_() -> Result<()> {
 				#[inline]
 				fn eq(&self, __other: &Self) -> bool {
 					match (self, __other) {
-						(Test { field: ref __field }, Test { field: ref __other_field }) =>
-							true && ::core::cmp::PartialEq::eq(__field, __other_field),
+						(Test { field: ref __field_field }, Test { field: ref __other_field_field }) =>
+							true && ::core::cmp::PartialEq::eq(__field_field, __other_field_field),
 					}
 				}
 			}
@@ -111,9 +111,9 @@ fn tuple() -> Result<()> {
 			impl<T> ::core::fmt::Debug for Test<T> {
 				fn fmt(&self, __f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
 					match self {
-						Test(ref __0) => {
+						Test(ref __field_0) => {
 							let mut __builder = ::core::fmt::Formatter::debug_tuple(__f, "Test");
-							::core::fmt::DebugTuple::field(&mut __builder, __0);
+							::core::fmt::DebugTuple::field(&mut __builder, __field_0);
 							::core::fmt::DebugTuple::finish(&mut __builder)
 						}
 					}
@@ -139,7 +139,7 @@ fn tuple() -> Result<()> {
 			impl<T> ::core::hash::Hash for Test<T> {
 				fn hash<__H: ::core::hash::Hasher>(&self, __state: &mut __H) {
 					match self {
-						Test(ref __0) => { ::core::hash::Hash::hash(__0, __state); }
+						Test(ref __field_0) => { ::core::hash::Hash::hash(__field_0, __state); }
 					}
 				}
 			}
@@ -148,8 +148,8 @@ fn tuple() -> Result<()> {
 				#[inline]
 				fn cmp(&self, __other: &Self) -> ::core::cmp::Ordering {
 					match (self, __other) {
-						(Test(ref __0), Test(ref __other_0)) =>
-							match ::core::cmp::Ord::cmp(__0, __other_0) {
+						(Test(ref __field_0), Test(ref __other_field_0)) =>
+							match ::core::cmp::Ord::cmp(__field_0, __other_field_0) {
 								::core::cmp::Ordering::Equal => ::core::cmp::Ordering::Equal,
 								__cmp => __cmp,
 							},
@@ -161,8 +161,8 @@ fn tuple() -> Result<()> {
 				#[inline]
 				fn eq(&self, __other: &Self) -> bool {
 					match (self, __other) {
-						(Test(ref __0), Test(ref __other_0)) =>
-							true && ::core::cmp::PartialEq::eq(__0, __other_0),
+						(Test(ref __field_0), Test(ref __other_field_0)) =>
+							true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 					}
 				}
 			}
@@ -204,13 +204,13 @@ fn enum_() -> Result<()> {
 	#[cfg(all(not(feature = "nightly"), feature = "safe"))]
 	let ord = quote! {
 		match self {
-			Test::A { field: ref __field } => ::core::cmp::Ordering::Less,
+			Test::A { field: ref __field_field } => ::core::cmp::Ordering::Less,
 			Test::B { } =>
 				match __other {
 					Test::A { .. } => ::core::cmp::Ordering::Greater,
 					_ => ::core::cmp::Ordering::Less,
 				},
-			Test::C(ref __0) =>
+			Test::C(ref __field_0) =>
 				match __other {
 					Test::A { .. } | Test::B { .. } => ::core::cmp::Ordering::Greater,
 					_ => ::core::cmp::Ordering::Less,
@@ -250,18 +250,18 @@ fn enum_() -> Result<()> {
 			impl<T> ::core::fmt::Debug for Test<T> {
 				fn fmt(&self, __f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
 					match self {
-						Test::A { field: ref __field } => {
+						Test::A { field: ref __field_field } => {
 							let mut __builder = ::core::fmt::Formatter::debug_struct(__f, "A");
-							::core::fmt::DebugStruct::field(&mut __builder, "field", __field);
+							::core::fmt::DebugStruct::field(&mut __builder, "field", __field_field);
 							::core::fmt::DebugStruct::finish(&mut __builder)
 						}
 						Test::B { } => {
 							let mut __builder = ::core::fmt::Formatter::debug_struct(__f, "B");
 							::core::fmt::DebugStruct::finish(&mut __builder)
 						}
-						Test::C(ref __0) => {
+						Test::C(ref __field_0) => {
 							let mut __builder = ::core::fmt::Formatter::debug_tuple(__f, "C");
-							::core::fmt::DebugTuple::field(&mut __builder, __0);
+							::core::fmt::DebugTuple::field(&mut __builder, __field_0);
 							::core::fmt::DebugTuple::finish(&mut __builder)
 						}
 						Test::D() => {
@@ -293,16 +293,16 @@ fn enum_() -> Result<()> {
 			impl<T> ::core::hash::Hash for Test<T> {
 				fn hash<__H: ::core::hash::Hasher>(&self, __state: &mut __H) {
 					match self {
-						Test::A { field: ref __field } => {
+						Test::A { field: ref __field_field } => {
 							::core::hash::Hash::hash(&::core::mem::discriminant(self), __state);
-							::core::hash::Hash::hash(__field, __state);
+							::core::hash::Hash::hash(__field_field, __state);
 						}
 						Test::B { } => {
 							::core::hash::Hash::hash(&::core::mem::discriminant(self), __state);
 						}
-						Test::C(ref __0) => {
+						Test::C(ref __field_0) => {
 							::core::hash::Hash::hash(&::core::mem::discriminant(self), __state);
-							::core::hash::Hash::hash(__0, __state);
+							::core::hash::Hash::hash(__field_0, __state);
 						}
 						Test::D() => {
 							::core::hash::Hash::hash(&::core::mem::discriminant(self), __state);
@@ -321,13 +321,13 @@ fn enum_() -> Result<()> {
 
 					if __self_disc == __other_disc {
 						match (self, __other) {
-							(Test::A { field: ref __field }, Test::A { field: ref __other_field }) =>
-								match ::core::cmp::Ord::cmp(__field, __other_field) {
+							(Test::A { field: ref __field_field }, Test::A { field: ref __other_field_field }) =>
+								match ::core::cmp::Ord::cmp(__field_field, __other_field_field) {
 									::core::cmp::Ordering::Equal => ::core::cmp::Ordering::Equal,
 									__cmp => __cmp,
 								},
-							(Test::C(ref __0), Test::C(ref __other_0)) =>
-								match ::core::cmp::Ord::cmp(__0, __other_0) {
+							(Test::C(ref __field_0), Test::C(ref __other_field_0)) =>
+								match ::core::cmp::Ord::cmp(__field_0, __other_field_0) {
 									::core::cmp::Ordering::Equal => ::core::cmp::Ordering::Equal,
 									__cmp => __cmp,
 								},
@@ -344,10 +344,10 @@ fn enum_() -> Result<()> {
 				fn eq(&self, __other: &Self) -> bool {
 					if ::core::mem::discriminant(self) == ::core::mem::discriminant(__other) {
 						match (self, __other) {
-							(Test::A { field: ref __field }, Test::A { field: ref __other_field }) =>
-								true && ::core::cmp::PartialEq::eq(__field, __other_field),
-							(Test::C(ref __0), Test::C(ref __other_0)) =>
-								true && ::core::cmp::PartialEq::eq(__0, __other_0),
+							(Test::A { field: ref __field_field }, Test::A { field: ref __other_field_field }) =>
+								true && ::core::cmp::PartialEq::eq(__field_field, __other_field_field),
+							(Test::C(ref __field_0), Test::C(ref __other_field_0)) =>
+								true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 							_ => true,
 						}
 					} else {

--- a/src/test/bound.rs
+++ b/src/test/bound.rs
@@ -17,7 +17,7 @@ fn bound() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test(ref __0, ref __1) => Test(::core::clone::Clone::clone(__0), ::core::clone::Clone::clone(__1)),
+						Test(ref __field_0, ref __field_1) => Test(::core::clone::Clone::clone(__field_0), ::core::clone::Clone::clone(__field_1)),
 					}
 				}
 			}
@@ -41,7 +41,7 @@ fn bound_multiple() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test(ref __0, ref __1) => Test(::core::clone::Clone::clone(__0), ::core::clone::Clone::clone(__1)),
+						Test(ref __field_0, ref __field_1) => Test(::core::clone::Clone::clone(__field_0), ::core::clone::Clone::clone(__field_1)),
 					}
 				}
 			}
@@ -63,7 +63,7 @@ fn custom_bound() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test(ref __0) => Test(::core::clone::Clone::clone(__0)),
+						Test(ref __field_0) => Test(::core::clone::Clone::clone(__field_0)),
 					}
 				}
 			}
@@ -87,7 +87,7 @@ fn where_() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test(ref __0, ref __1) => Test(::core::clone::Clone::clone(__0), ::core::clone::Clone::clone(__1)),
+						Test(ref __field_0, ref __field_1) => Test(::core::clone::Clone::clone(__field_0), ::core::clone::Clone::clone(__field_1)),
 					}
 				}
 			}
@@ -109,7 +109,7 @@ fn associated_type() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test(ref __0) => Test(::core::clone::Clone::clone(__0)),
+						Test(ref __field_0) => Test(::core::clone::Clone::clone(__field_0)),
 					}
 				}
 			}
@@ -131,7 +131,7 @@ fn associated_type_custom_bound() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test(ref __0) => Test(::core::clone::Clone::clone(__0)),
+						Test(ref __field_0) => Test(::core::clone::Clone::clone(__field_0)),
 					}
 				}
 			}
@@ -165,10 +165,10 @@ fn check_trait_bounds() -> Result<()> {
 			{
 				fn fmt(&self, __f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
 					match self {
-						Test(ref __0, ref __1) => {
+						Test(ref __field_0, ref __field_1) => {
 							let mut __builder = ::core::fmt::Formatter::debug_tuple(__f, "Test");
-							::core::fmt::DebugTuple::field(&mut __builder, __0);
-							::core::fmt::DebugTuple::field(&mut __builder, __1);
+							::core::fmt::DebugTuple::field(&mut __builder, __field_0);
+							::core::fmt::DebugTuple::field(&mut __builder, __field_1);
 							::core::fmt::DebugTuple::finish(&mut __builder)
 						}
 					}
@@ -201,9 +201,9 @@ fn check_trait_bounds() -> Result<()> {
 			{
 				fn hash<__H: ::core::hash::Hasher>(&self, __state: &mut __H) {
 					match self {
-						Test(ref __0, ref __1) => {
-							::core::hash::Hash::hash(__0, __state);
-							::core::hash::Hash::hash(__1, __state);
+						Test(ref __field_0, ref __field_1) => {
+							::core::hash::Hash::hash(__field_0, __state);
+							::core::hash::Hash::hash(__field_1, __state);
 						}
 					}
 				}
@@ -215,9 +215,9 @@ fn check_trait_bounds() -> Result<()> {
 				#[inline]
 				fn cmp(&self, __other: &Self) -> ::core::cmp::Ordering {
 					match (self, __other) {
-						(Test(ref __0, ref __1), Test(ref __other_0, ref __other_1)) =>
-							match ::core::cmp::Ord::cmp(__0, __other_0) {
-								::core::cmp::Ordering::Equal => match ::core::cmp::Ord::cmp(__1, __other_1) {
+						(Test(ref __field_0, ref __field_1), Test(ref __other_field_0, ref __other_field_1)) =>
+							match ::core::cmp::Ord::cmp(__field_0, __other_field_0) {
+								::core::cmp::Ordering::Equal => match ::core::cmp::Ord::cmp(__field_1, __other_field_1) {
 									::core::cmp::Ordering::Equal => ::core::cmp::Ordering::Equal,
 									__cmp => __cmp,
 								},
@@ -233,10 +233,10 @@ fn check_trait_bounds() -> Result<()> {
 				#[inline]
 				fn eq(&self, __other: &Self) -> bool {
 					match (self, __other) {
-						(Test(ref __0, ref __1), Test(ref __other_0, ref __other_1)) =>
+						(Test(ref __field_0, ref __field_1), Test(ref __other_field_0, ref __other_field_1)) =>
 							true
-							&& ::core::cmp::PartialEq::eq(__0, __other_0)
-							&& ::core::cmp::PartialEq::eq(__1, __other_1),
+							&& ::core::cmp::PartialEq::eq(__field_0, __other_field_0)
+							&& ::core::cmp::PartialEq::eq(__field_1, __other_field_1),
 					}
 				}
 			}
@@ -285,10 +285,10 @@ fn check_multiple_trait_bounds() -> Result<()> {
 			{
 				fn fmt(&self, __f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
 					match self {
-						Test(ref __0, ref __1) => {
+						Test(ref __field_0, ref __field_1) => {
 							let mut __builder = ::core::fmt::Formatter::debug_tuple(__f, "Test");
-							::core::fmt::DebugTuple::field(&mut __builder, __0);
-							::core::fmt::DebugTuple::field(&mut __builder, __1);
+							::core::fmt::DebugTuple::field(&mut __builder, __field_0);
+							::core::fmt::DebugTuple::field(&mut __builder, __field_1);
 							::core::fmt::DebugTuple::finish(&mut __builder)
 						}
 					}
@@ -327,9 +327,9 @@ fn check_multiple_trait_bounds() -> Result<()> {
 			{
 				fn hash<__H: ::core::hash::Hasher>(&self, __state: &mut __H) {
 					match self {
-						Test(ref __0, ref __1) => {
-							::core::hash::Hash::hash(__0, __state);
-							::core::hash::Hash::hash(__1, __state);
+						Test(ref __field_0, ref __field_1) => {
+							::core::hash::Hash::hash(__field_0, __state);
+							::core::hash::Hash::hash(__field_1, __state);
 						}
 					}
 				}
@@ -343,9 +343,9 @@ fn check_multiple_trait_bounds() -> Result<()> {
 				#[inline]
 				fn cmp(&self, __other: &Self) -> ::core::cmp::Ordering {
 					match (self, __other) {
-						(Test(ref __0, ref __1), Test(ref __other_0, ref __other_1)) =>
-							match ::core::cmp::Ord::cmp(__0, __other_0) {
-								::core::cmp::Ordering::Equal => match ::core::cmp::Ord::cmp(__1, __other_1) {
+						(Test(ref __field_0, ref __field_1), Test(ref __other_field_0, ref __other_field_1)) =>
+							match ::core::cmp::Ord::cmp(__field_0, __other_field_0) {
+								::core::cmp::Ordering::Equal => match ::core::cmp::Ord::cmp(__field_1, __other_field_1) {
 									::core::cmp::Ordering::Equal => ::core::cmp::Ordering::Equal,
 									__cmp => __cmp,
 								},
@@ -363,10 +363,10 @@ fn check_multiple_trait_bounds() -> Result<()> {
 				#[inline]
 				fn eq(&self, __other: &Self) -> bool {
 					match (self, __other) {
-						(Test(ref __0, ref __1), Test(ref __other_0, ref __other_1)) =>
+						(Test(ref __field_0, ref __field_1), Test(ref __other_field_0, ref __other_field_1)) =>
 							true
-							&& ::core::cmp::PartialEq::eq(__0, __other_0)
-							&& ::core::cmp::PartialEq::eq(__1, __other_1),
+							&& ::core::cmp::PartialEq::eq(__field_0, __other_field_0)
+							&& ::core::cmp::PartialEq::eq(__field_1, __other_field_1),
 					}
 				}
 			}

--- a/src/test/clone.rs
+++ b/src/test/clone.rs
@@ -15,7 +15,7 @@ fn struct_() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test { field: ref __field } => Test { field: ::core::clone::Clone::clone(__field) },
+						Test { field: ref __field_field } => Test { field: ::core::clone::Clone::clone(__field_field) },
 					}
 				}
 			}
@@ -35,7 +35,7 @@ fn tuple() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test(ref __0) => Test(::core::clone::Clone::clone(__0)),
+						Test(ref __field_0) => Test(::core::clone::Clone::clone(__field_0)),
 					}
 				}
 			}
@@ -61,9 +61,9 @@ fn enum_() -> Result<()> {
 				#[inline]
 				fn clone(&self) -> Self {
 					match self {
-						Test::A { field: ref __field } => Test::A { field: ::core::clone::Clone::clone(__field) },
+						Test::A { field: ref __field_field } => Test::A { field: ::core::clone::Clone::clone(__field_field) },
 						Test::B { } => Test::B { },
-						Test::C(ref __0) => Test::C(::core::clone::Clone::clone(__0)),
+						Test::C(ref __field_0) => Test::C(::core::clone::Clone::clone(__field_0)),
 						Test::D() => Test::D(),
 						Test::E => Test::E,
 					}

--- a/src/test/enum_.rs
+++ b/src/test/enum_.rs
@@ -82,8 +82,8 @@ fn one_data() -> Result<()> {
 				#[inline]
 				fn eq(&self, __other: &Self) -> bool {
 					match (self, __other) {
-						(Test::A(ref __0), Test::A(ref __other_0)) =>
-							true && ::core::cmp::PartialEq::eq(__0, __other_0),
+						(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+							true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 					}
 				}
 			}
@@ -92,8 +92,8 @@ fn one_data() -> Result<()> {
 				#[inline]
 				fn partial_cmp(&self, __other: &Self) -> ::core::option::Option<::core::cmp::Ordering> {
 					match (self, __other) {
-						(Test::A(ref __0), Test::A(ref __other_0)) =>
-							match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+						(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+							match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 								::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 								__cmp => __cmp,
 							},
@@ -134,8 +134,8 @@ fn two_data() -> Result<()> {
 	#[cfg(all(not(feature = "nightly"), feature = "safe"))]
 	let partial_ord = quote! {
 		match self {
-			Test::A(ref __0) => ::core::option::Option::Some(::core::cmp::Ordering::Less),
-			Test::B(ref __0) => ::core::option::Option::Some(::core::cmp::Ordering::Greater),
+			Test::A(ref __field_0) => ::core::option::Option::Some(::core::cmp::Ordering::Less),
+			Test::B(ref __field_0) => ::core::option::Option::Some(::core::cmp::Ordering::Greater),
 		}
 	};
 
@@ -150,10 +150,10 @@ fn two_data() -> Result<()> {
 				fn eq(&self, __other: &Self) -> bool {
 					if ::core::mem::discriminant(self) == ::core::mem::discriminant(__other) {
 						match (self, __other) {
-							(Test::A(ref __0), Test::A(ref __other_0)) =>
-								true && ::core::cmp::PartialEq::eq(__0, __other_0),
-							(Test::B(ref __0), Test::B(ref __other_0)) =>
-								true && ::core::cmp::PartialEq::eq(__0, __other_0),
+							(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+								true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
+							(Test::B(ref __field_0), Test::B(ref __other_field_0)) =>
+								true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 							_ => #unreachable,
 						}
 					} else {
@@ -169,13 +169,13 @@ fn two_data() -> Result<()> {
 
 					if __self_disc == __other_disc {
 						match (self, __other) {
-							(Test::A(ref __0), Test::A(ref __other_0)) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+							(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
 								},
-							(Test::B(ref __0), Test::B(ref __other_0)) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+							(Test::B(ref __field_0), Test::B(ref __other_field_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
 								},
@@ -216,7 +216,7 @@ fn unit() -> Result<()> {
 	#[cfg(all(not(feature = "nightly"), feature = "safe"))]
 	let partial_ord = quote! {
 		match self {
-			Test::A(ref __0) => ::core::option::Option::Some(::core::cmp::Ordering::Less),
+			Test::A(ref __field_0) => ::core::option::Option::Some(::core::cmp::Ordering::Less),
 			Test::B => ::core::option::Option::Some(::core::cmp::Ordering::Greater),
 		}
 	};
@@ -232,8 +232,8 @@ fn unit() -> Result<()> {
 				fn eq(&self, __other: &Self) -> bool {
 					if ::core::mem::discriminant(self) == ::core::mem::discriminant(__other) {
 						match (self, __other) {
-							(Test::A(ref __0), Test::A(ref __other_0)) =>
-								true && ::core::cmp::PartialEq::eq(__0, __other_0),
+							(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+								true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 							_ => true,
 						}
 					} else {
@@ -249,8 +249,8 @@ fn unit() -> Result<()> {
 
 					if __self_disc == __other_disc {
 						match (self, __other) {
-							(Test::A(ref __0), Test::A(ref __other_0)) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+							(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
 								},
@@ -291,7 +291,7 @@ fn struct_unit() -> Result<()> {
 	#[cfg(all(not(feature = "nightly"), feature = "safe"))]
 	let partial_ord = quote! {
 		match self {
-			Test::A(ref __0) => ::core::option::Option::Some(::core::cmp::Ordering::Less),
+			Test::A(ref __field_0) => ::core::option::Option::Some(::core::cmp::Ordering::Less),
 			Test::B { } => ::core::option::Option::Some(::core::cmp::Ordering::Greater),
 		}
 	};
@@ -307,8 +307,8 @@ fn struct_unit() -> Result<()> {
 				fn eq(&self, __other: &Self) -> bool {
 					if ::core::mem::discriminant(self) == ::core::mem::discriminant(__other) {
 						match (self, __other) {
-							(Test::A(ref __0), Test::A(ref __other_0)) =>
-								true && ::core::cmp::PartialEq::eq(__0, __other_0),
+							(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+								true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 							_ => true,
 						}
 					} else {
@@ -324,8 +324,8 @@ fn struct_unit() -> Result<()> {
 
 					if __self_disc == __other_disc {
 						match (self, __other) {
-							(Test::A(ref __0), Test::A(ref __other_0)) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+							(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
 								},
@@ -366,7 +366,7 @@ fn tuple_unit() -> Result<()> {
 	#[cfg(all(not(feature = "nightly"), feature = "safe"))]
 	let partial_ord = quote! {
 		match self {
-			Test::A(ref __0) => ::core::option::Option::Some(::core::cmp::Ordering::Less),
+			Test::A(ref __field_0) => ::core::option::Option::Some(::core::cmp::Ordering::Less),
 			Test::B() => ::core::option::Option::Some(::core::cmp::Ordering::Greater),
 		}
 	};
@@ -382,8 +382,8 @@ fn tuple_unit() -> Result<()> {
 				fn eq(&self, __other: &Self) -> bool {
 					if ::core::mem::discriminant(self) == ::core::mem::discriminant(__other) {
 						match (self, __other) {
-							(Test::A(ref __0), Test::A(ref __other_0)) =>
-								true && ::core::cmp::PartialEq::eq(__0, __other_0),
+							(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+								true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 							_ => true,
 						}
 					} else {
@@ -399,8 +399,8 @@ fn tuple_unit() -> Result<()> {
 
 					if __self_disc == __other_disc {
 						match (self, __other) {
-							(Test::A(ref __0), Test::A(ref __other_0)) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+							(Test::A(ref __field_0), Test::A(ref __other_field_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
 								},

--- a/src/test/incomparable.rs
+++ b/src/test/incomparable.rs
@@ -25,7 +25,7 @@ fn variants() -> Result<()> {
 				fn eq(&self, __other: &Self) -> bool {
 					if ::core::mem::discriminant(self) == ::core::mem::discriminant(__other) {
 						match (self, __other) {
-							(Test::C(ref __0), Test::C(ref __other_0)) => true && ::core::cmp::PartialEq::eq(__0, __other_0),
+							(Test::C(ref __field_0), Test::C(ref __other_field_0)) => true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 							(Test::B(..) | Test::D | Test::E {..}, ..) => false,
 							_ => true,
 						}
@@ -46,8 +46,8 @@ fn variants() -> Result<()> {
 					let __other_disc = ::core::mem::discriminant(__other);
 					if __self_disc == __other_disc {
 						match (self, __other) {
-							(Test::C(ref __0), Test::C(ref __other_0)) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+							(Test::C(ref __field_0), Test::C(ref __other_field_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) =>
 										::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
@@ -244,8 +244,8 @@ fn enum_non_empty_and_incomparable_variant() -> Result<()> {
 				fn eq(&self, __other: &Self) -> bool {
 					if ::core::mem::discriminant(self) == ::core::mem::discriminant(__other) {
 						match (self, __other) {
-							(Test::B(ref __0), Test::B(ref __other_0)) =>
-								true && ::core::cmp::PartialEq::eq(__0, __other_0),
+							(Test::B(ref __field_0), Test::B(ref __other_field_0)) =>
+								true && ::core::cmp::PartialEq::eq(__field_0, __other_field_0),
 							(Test::A, ..) => false,
 							_ => unsafe { ::core::hint::unreachable_unchecked() },
 						}
@@ -261,8 +261,8 @@ fn enum_non_empty_and_incomparable_variant() -> Result<()> {
 						::core::option::Option::None
 					} else {
 						match (self, __other) {
-							(Test::B(ref __0), Test::B(ref __other_0)) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+							(Test::B(ref __field_0), Test::B(ref __other_field_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) =>
 										::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,

--- a/src/test/partial_ord.rs
+++ b/src/test/partial_ord.rs
@@ -15,8 +15,8 @@ fn struct_() -> Result<()> {
 				#[inline]
 				fn partial_cmp(&self, __other: &Self) -> ::core::option::Option<::core::cmp::Ordering> {
 					match (self, __other) {
-						(Test { field: ref __field }, Test { field: ref __other_field }) =>
-							match ::core::cmp::PartialOrd::partial_cmp(__field, __other_field) {
+						(Test { field: ref __field_field }, Test { field: ref __other_field_field }) =>
+							match ::core::cmp::PartialOrd::partial_cmp(__field_field, __other_field_field) {
 								::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 								__cmp => __cmp,
 							},
@@ -39,8 +39,8 @@ fn tuple() -> Result<()> {
 				#[inline]
 				fn partial_cmp(&self, __other: &Self) -> ::core::option::Option<::core::cmp::Ordering> {
 					match (self, __other) {
-						(Test(ref __0), Test(ref __other_0)) =>
-							match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+						(Test(ref __field_0), Test(ref __other_field_0)) =>
+							match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 								::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 								__cmp => __cmp,
 							},
@@ -77,13 +77,13 @@ fn enum_() -> Result<()> {
 	#[cfg(all(not(feature = "nightly"), feature = "safe"))]
 	let partial_ord = quote! {
 		match self {
-			Test::A { field: ref __field } => ::core::option::Option::Some(::core::cmp::Ordering::Less),
+			Test::A { field: ref __field_field } => ::core::option::Option::Some(::core::cmp::Ordering::Less),
 			Test::B { } =>
 				match __other {
 					Test::A { .. } => ::core::option::Option::Some(::core::cmp::Ordering::Greater),
 					_ => ::core::option::Option::Some(::core::cmp::Ordering::Less),
 				},
-			Test::C(ref __0) =>
+			Test::C(ref __field_0) =>
 				match __other {
 					Test::A { .. } | Test::B { .. } => ::core::option::Option::Some(::core::cmp::Ordering::Greater),
 					_ => ::core::option::Option::Some(::core::cmp::Ordering::Less),
@@ -116,13 +116,13 @@ fn enum_() -> Result<()> {
 
 					if __self_disc == __other_disc {
 						match (self, __other) {
-							(Test::A { field: ref __field }, Test::A { field: ref __other_field }) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__field, __other_field) {
+							(Test::A { field: ref __field_field }, Test::A { field: ref __other_field_field }) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_field, __other_field_field) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
 								},
-							(Test::C(ref __0), Test::C(ref __other_0)) =>
-								match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
+							(Test::C(ref __field_0), Test::C(ref __other_field_0)) =>
+								match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
 								},
@@ -176,9 +176,9 @@ fn bound() -> Result<()> {
 				#[inline]
 				fn cmp(&self, __other: &Self) -> ::core::cmp::Ordering {
 					match (self, __other) {
-						(Test(ref __0, ref __1), Test(ref __other_0, ref __other_1)) =>
-							match ::core::cmp::Ord::cmp(__0, __other_0) {
-								::core::cmp::Ordering::Equal => match ::core::cmp::Ord::cmp(__1, __other_1) {
+						(Test(ref __field_0, ref __field_1), Test(ref __other_field_0, ref __other_field_1)) =>
+							match ::core::cmp::Ord::cmp(__field_0, __other_field_0) {
+								::core::cmp::Ordering::Equal => match ::core::cmp::Ord::cmp(__field_1, __other_field_1) {
 									::core::cmp::Ordering::Equal => ::core::cmp::Ordering::Equal,
 									__cmp => __cmp,
 								},
@@ -192,9 +192,9 @@ fn bound() -> Result<()> {
 				#[inline]
 				fn partial_cmp(&self, __other: &Self) -> ::core::option::Option<::core::cmp::Ordering> {
 					match (self, __other) {
-						(Test(ref __0, ref __1), Test(ref __other_0, ref __other_1)) =>
-							match ::core::cmp::PartialOrd::partial_cmp(__0, __other_0) {
-								::core::option::Option::Some(::core::cmp::Ordering::Equal) => match ::core::cmp::PartialOrd::partial_cmp(__1, __other_1) {
+						(Test(ref __field_0, ref __field_1), Test(ref __other_field_0, ref __other_field_1)) =>
+							match ::core::cmp::PartialOrd::partial_cmp(__field_0, __other_field_0) {
+								::core::option::Option::Some(::core::cmp::Ordering::Equal) => match ::core::cmp::PartialOrd::partial_cmp(__field_1, __other_field_1) {
 									::core::option::Option::Some(::core::cmp::Ordering::Equal) => ::core::option::Option::Some(::core::cmp::Ordering::Equal),
 									__cmp => __cmp,
 								},

--- a/src/test/skip.rs
+++ b/src/test/skip.rs
@@ -15,7 +15,7 @@ fn struct_inner() -> Result<()> {
 			impl<T> ::core::fmt::Debug for Test<T> {
 				fn fmt(&self, __f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
 					match self {
-						Test(ref __0) => {
+						Test(ref __field_0) => {
 							let mut __builder = ::core::fmt::Formatter::debug_tuple(__f, "Test");
 							::core::fmt::DebugTuple::finish(&mut __builder)
 						}
@@ -40,7 +40,7 @@ fn enum_inner() -> Result<()> {
 			impl<T> ::core::fmt::Debug for Test<T> {
 				fn fmt(&self, __f: &mut ::core::fmt::Formatter<'_>) -> ::core::fmt::Result {
 					match self {
-						Test::A(ref __0) => {
+						Test::A(ref __field_0) => {
 							let mut __builder = ::core::fmt::Formatter::debug_tuple(__f, "A");
 							::core::fmt::DebugTuple::finish(&mut __builder)
 						}
@@ -118,8 +118,8 @@ fn variants_empty() -> Result<()> {
 	#[cfg(all(not(feature = "nightly"), feature = "safe"))]
 	let ord = quote! {
 		match self {
-			Test::A(ref __0) => ::core::cmp::Ordering::Less,
-			Test::B(ref __0) => ::core::cmp::Ordering::Greater,
+			Test::A(ref __field_0) => ::core::cmp::Ordering::Less,
+			Test::B(ref __field_0) => ::core::cmp::Ordering::Greater,
 		}
 	};
 
@@ -177,8 +177,8 @@ fn variants_partly_empty() -> Result<()> {
 	#[cfg(all(not(feature = "nightly"), feature = "safe"))]
 	let ord = quote! {
 		match self {
-			Test::A(ref __0) => ::core::cmp::Ordering::Less,
-			Test::B(ref __0, ref __1) => ::core::cmp::Ordering::Greater,
+			Test::A(ref __field_0) => ::core::cmp::Ordering::Less,
+			Test::B(ref __field_0, ref __field_1) => ::core::cmp::Ordering::Greater,
 		}
 	};
 
@@ -199,8 +199,8 @@ fn variants_partly_empty() -> Result<()> {
 
 					if __self_disc == __other_disc {
 						match (self , __other) {
-							(Test::B(ref __0, ref __1), Test::B(ref __other_0, ref __other_1)) =>
-								match ::core::cmp::Ord::cmp(__1 ,__other_1) {
+							(Test::B(ref __field_0, ref __field_1), Test::B(ref __other_field_0, ref __other_field_1)) =>
+								match ::core::cmp::Ord::cmp(__field_1 ,__other_field_1) {
 									::core::cmp::Ordering::Equal => ::core::cmp::Ordering::Equal, __cmp => __cmp,
 								},
 							_ => ::core::cmp::Ordering::Equal,

--- a/src/test/zeroize.rs
+++ b/src/test/zeroize.rs
@@ -16,8 +16,8 @@ fn basic() -> Result<()> {
 					use ::zeroize::Zeroize;
 
 					match self {
-						Test(ref mut __0) => {
-							__0.zeroize();
+						Test(ref mut __field_0) => {
+							__field_0.zeroize();
 						}
 					}
 				}
@@ -53,9 +53,9 @@ fn drop() -> Result<()> {
 					use ::zeroize::__internal::AssertZeroizeOnDrop;
 
 					match self {
-						Test(ref mut __0, ref mut __1) => {
-							__0.zeroize_or_on_drop();
-							__1.zeroize_or_on_drop();
+						Test(ref mut __field_0, ref mut __field_1) => {
+							__field_0.zeroize_or_on_drop();
+							__field_1.zeroize_or_on_drop();
 						}
 					}
 				}
@@ -90,9 +90,9 @@ fn both() -> Result<()> {
 				use ::zeroize::__internal::AssertZeroizeOnDrop;
 
 				match self {
-					Test(ref mut __0, ref mut __1) => {
-						__0.zeroize_or_on_drop();
-						__1.zeroize_or_on_drop();
+					Test(ref mut __field_0, ref mut __field_1) => {
+						__field_0.zeroize_or_on_drop();
+						__field_1.zeroize_or_on_drop();
 					}
 				}
 			}
@@ -116,9 +116,9 @@ fn both() -> Result<()> {
 					use ::zeroize::Zeroize;
 
 					match self {
-						Test(ref mut __0, ref mut __1) => {
-							__0.zeroize();
-							__1.zeroize();
+						Test(ref mut __field_0, ref mut __field_1) => {
+							__field_0.zeroize();
+							__field_1.zeroize();
 						}
 					}
 				}
@@ -144,8 +144,8 @@ fn crate_() -> Result<()> {
 					use zeroize_::Zeroize;
 
 					match self {
-						Test(ref mut __0) => {
-							__0.zeroize();
+						Test(ref mut __field_0) => {
+							__field_0.zeroize();
 						}
 					}
 				}
@@ -181,8 +181,8 @@ fn crate_drop() -> Result<()> {
 					use zeroize_::__internal::AssertZeroizeOnDrop;
 
 					match self {
-						Test(ref mut __0) => {
-							__0.zeroize_or_on_drop();
+						Test(ref mut __field_0) => {
+							__field_0.zeroize_or_on_drop();
 						}
 					}
 				}
@@ -208,8 +208,8 @@ fn fqs() -> Result<()> {
 					use ::zeroize::Zeroize;
 
 					match self {
-						Test(ref mut __0) => {
-							::zeroize::Zeroize::zeroize(__0);
+						Test(ref mut __field_0) => {
+							::zeroize::Zeroize::zeroize(__field_0);
 						}
 					}
 				}

